src/urllib3/contrib/pyopenssl.py

```diff
@@ -380,14 +380,15 @@ def shutdown(self) -> None:
         self.connection.shutdown()
 
     def close(self) -> None:
-        if self._io_refs < 1:
-            try:
-                self._closed = True
-                return self.connection.close()  # type: ignore[no-any-return]
-            except OpenSSL.SSL.Error:
-                return
-        else:
-            self._io_refs -= 1
+        self._closed = True
+        if self._io_refs <= 0:
+            self._real_close()
+
+    def _real_close(self) -> None:
```

Comment:
If `real_close` is only used once, do we need to separate it into its own function? Also `_io_refs` is no longer decremented, is that expected?

Reply:
I'm copying the pattern from the stdlib socket